### ISSUE TITLE
AMPds Data can't run 

### ISSUE THREAD

**Issue by Wusir2018** [nilmtk/nilmtk-contrib#24]
Hello everyone,
I can run the algorithm with REDD. When I use AMPds data to test the algorithm, I get the error.  There are pictures about my code and error. I think the timestamp of AMPds  may cause this problem, but I don't know how to solve it.
Could you help me?
Thank you very much!
`code:`
![1](https://user-images.githubusercontent.com/51500032/74580413-08133800-4fdf-11ea-832e-98b13eff5d30.PNG)
![2](https://user-images.githubusercontent.com/51500032/74580416-0d708280-4fdf-11ea-8fe7-d6f01b0698e2.PNG)

`error:`
![3](https://user-images.githubusercontent.com/51500032/74580421-1d886200-4fdf-11ea-94cf-42b7429c4b37.PNG)
![4](https://user-images.githubusercontent.com/51500032/74580422-211be900-4fdf-11ea-8322-5c36db089232.PNG)

**Comment by eddie-LA** [comment]
I have the same problem...

**Comment by levaphenyl** [comment]
@eddie-LA Thank you for the analysis! This looks like a bug in the class `API` indeed.
The design question there is the expected output of the API. What is the metric value when predicting 3 different time series?

The current approach is to concatenate all time series, also in the regressor implementation of `nilmtk-contrib`. However this mixes values from different units (W, VA, etc.) which has little physical meaning.
Keeping the channels separated could be an interesting approach, i.e. feeding matrices of shape (N, 3) instead of column vectors of shape (3 * N, 1) in the models. But then, how do you define the metrics? Do you expect 3 different metrics at the end of the experiment?

I see 2 cases for using the `API` with multiple channels:
1. you are really interested in data fusion of active and reactive power and the effects on algorithmic performance, or
2. you are using datasets where the mains were measured with a different unit than the appliances but you are only interested in the reactive power.

The API should handle both, but fixing case 2. is the most urgent  in my opinion.

**Comment by eddie-LA** [comment]
I was mainly interested in 1 in my case. I wanted to experiment with the algorithms in NILMTK and how they fare against Makonin's Temporal Convolutional architecture [here](http://makonin.com/doc/ICASSP_2019.pdf). He shows that reactive power is more useful as a disaggregating metric but since customers are billed for active power, it seems counterproductive to disregard it. It has its uses and the best results were recorded when the algorithm has access to several signatures anyway.

**Comment by levaphenyl** [comment]
Interesting research!
If I understand it well, you need `NILMTK` and `nilmtk-contrib` to accept active and reactive power as input. As output the algorithms in your research are only infering the active power. I that right?
We thus have multi-channel in and single-channel out.

**Comment by eddie-LA** [comment]
Correct! Active power is the only usable output metric, but not the only usable input metric.
